### PR TITLE
Fixed crash when freeing unused pointers

### PR DESCRIPTION
**Pull request by boris-martin**
When one interface doesn't write data, an unitialized pointer gets freed in the end. This PR initalizes it to NULL when necessary.
Example to reproduce: https://github.com/precice/tutorials/pull/271.

### CODE REVIEW

File: adapter/PreciceInterface.c [hunk 1]
```diff
@@ -700,6 +700,8 @@ void PreciceInterface_ConfigureCouplingData(PreciceInterface *interface, Simulat
   interface->numWriteData = config->numWriteData;
   if (config->numWriteData > 0)
     interface->writeData = malloc(config->numWriteData * sizeof(int));
+  else
+    interface->writeData = NULL;
```
Comment (MakisH) [inline adapter/PreciceInterface.c:704]: I am not familiar with the codebase or sure about good practices in C, but I understand that will make sure that the free at line 769 is not triggered when no write-data is configured. I guess one could default this to `NULL` in the structure definition.

Shouldn't the same also apply to `readData`?

Applying this change does not fix the issue for me, neither in debug nor in release mode. Does this actually work for you? Does it work on the Demo VM or in some CI environment?

File: tools/format-code.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 
 # Format all adapter C and C++ files with clang-format
-find ./ \( -iname "*.h" -o -iname "*.hpp" -o -iname "*.c" -o -iname "*.cpp" \) -exec clang-format -i {} \;
+find ./ \( -iname "*.h" -o -iname "*.hpp" -o -iname "*.c" -o -iname "*.cpp" \) -exec clang-format-11 -i {} \;
```
[the remark this answers is not in the document]
Comment (boris-martin) [reply tools/format-code.sh:4]: Actually an another side-effect of using "git commit -a" at night, but we can keep it yes

File: adapter/PreciceInterface.c (hunk 1 above)
Comment (boris-martin) [reply adapter/PreciceInterface.c:704]: To solve the error, I needed this PR and some changes to the tutorial case that I just pushed (I had forgotten yesterday): CalculiX time & preCICE end time were longer than OpenFOAM end time.
Does it work for you with this PR + updated tutorial?

(Might be interesting to generate better errors when time don't match, still).

I moved the change to initialization and also initialized `readData`.

**Review by MakisH — APPROVED** [review]
The simulation completes now and moving to the default value makes much more sense! :+1:

**Pull request merged**